### PR TITLE
Support for PERIOD type

### PR DESCRIPTION
**Pull request by Katakam-Phani-Mahidhar**

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java [hunk 1]
```diff
@@ -11315,6 +11336,37 @@ private RelNode createRelNodeWithQualifyStatement() {
...
   }
 
+  @Test public void testPeriodValueFunctions() {
+    final RelBuilder builder = relBuilder();
+    final RexNode date = builder.literal(new DateString("2000-01-01"));
+    final RexNode periodConstructor = builder.call(PERIOD_CONSTRUCTOR, date, date);
+    RelNode root = builder.scan("EMP").project(periodConstructor).build();
+    final String teradataPeriod = "SELECT PERIOD(DATE '2000-01-01', DATE '2000-01-01') AS \"$f0\""
+        + "\nFROM \"scott\".\"EMP\"";
+    final String bigQueryPeriod = "SELECT RANGE(DATE '2000-01-01', DATE '2000-01-01') AS `$f0`"
```
Comment (Abhi-210793) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java:11346]: Please use two different dates

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java [hunk 2]
```diff
@@ -10296,6 +10298,25 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
     assertThat(toSql(root, DatabaseProduct.BIG_QUERY.getDialect()), isLinux(expectedBiqQuery));
   }
 
+  @Test public void testBQRangeLiteral() {
+    final RelBuilder builder = relBuilder();
+    final RexNode upperRange =
+        builder.getRexBuilder().makeDateLiteral(new DateString("2000-12-12"));
```
Comment (Abhi-210793) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java:10304]: Please use two different dates

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 3]
```diff
@@ -993,6 +1004,26 @@ static RelDataType deriveTypeSplit(SqlOperatorBinding operatorBinding,
       new SqlCurrentTimestampFunction("CURRENT_TIMESTAMP_LTZ",
           SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE);
 
+  @LibraryOperator(libraries = {BIG_QUERY})
+  public static final SqlFunction RANGE_LITERAL =
```
[the remark this answers is not in the document]
Comment (Katakam-Phani-Mahidhar) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java:1013]: this is an operator for representing RANGE literals in bq, there is a PERIOD_CONSTRUCTOR already which is equivalent to RANGE function.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java (hunk 2 above)
Comment (Katakam-Phani-Mahidhar) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java:10304]: changed

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java (hunk 1 above)
Comment (Katakam-Phani-Mahidhar) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java]: changed

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Abhi-210793) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java:1013]: can we connect to discuss this

**Comment by Abhi-210793** [comment]
Review done

**Review by dm-shantanuc — APPROVED** [review]

**Pull request merged**